Foundation/NSXMLParser.swift

```diff
@@ -73,48 +73,40 @@ internal func _NSXMLParserExternalEntityWithURL(interface: _CFXMLInterface, urlS
     
     switch policy {
     case .ResolveExternalEntitiesSameOriginOnly:
-        if let url = parser._url {
-            if a == nil {
-                a = NSURL(string: String(urlStr))
+        guard let url = parser._url else { break }
+        
+        if a == nil {
+            a = NSURL(string: String(urlStr))
+        }
+        
+        guard let aUrl = a else { break }
```

Comment:
why don't you just return NSURL(string: String(urlStr)) in the guard's else case instead of checking for a's (non)existence twice?